toolium/behave/environment.py

```diff
@@ -83,6 +83,9 @@ def before_feature(context, feature):
     # Behave dynamic environment
     context.dyn_env.get_steps_from_feature_description(feature.description)
     context.dyn_env.execute_before_feature_steps(context)
+    if context.dyn_env.feature_error:
+        # Mark this Scenario as skipped. Steps will not be executed.
+        feature.mark_skipped()
```

Comment:
I think that this operation (skipped) should be included into the `DynamicEnvironment` class, like this:

```
environment.py current
--------------------------
   context.dyn_env.execute_before_feature_steps(context)
   if context.dyn_env.feature_error:
       # Mark this Scenario as skipped. Steps will not be executed.
       feature.mark_skipped()
```    
   **Replace by:**
```
   env_utils.py
  -----------------    
    def execute_before_feature_steps(self, context):
       """            
        actions before the feature
       param context: It’s a clever place where you and behave can store information to share around, automatically managed by behave.
       """          
        self.__execute_steps_by_action(context, ACTIONS_BEFORE_FEATURE)
        if self.feature_error:
            # Mark this Scenario as skipped. Steps will not be executed.
            feature.mark_skipped()
        return feature


    environment.py
-------------------------
    # Mark this Feature as skipped if some step is failed and the remaining steps will not be executed.
    feature = context.dyn_env.execute_before_feature_steps(context, feature)
```

The same behavior in `before_scenario`, etc.

Reply:
all implemented logic is now part of `env_utils.py`